src/Editor/EditorSupport.php

```diff
@@ -91,6 +91,6 @@ function () {
 	 * @return bool
 	 */
 	public function editor_supports_amp_block_editor_features() {
-		return $this->dependency_support->has_support();
+		return $this->dependency_support->has_support() && use_block_editor_for_post( get_post() );
```

Comment:
This might be too low level as it would prevent using this method in other contexts, such as if we want to show a notice on the settings screen or site health to encourage the user to update to use the block editor.

Reply:
That would be what `DependencySupport::has_support()` is for no? `EditorSupport::editor_supports_amp_block_editor_features()` is specifically for checking block editor support and is currently being used as a check before enqueuing block editor related assets.

Reply:
I don't think so because isn't the problem that we're enqueueing on the wrong screen?

It's being used in three places:

1. `\AMP_Post_Meta_Box::enqueue_block_assets()`
2. `\AMP_Validation_Manager::enqueue_block_validation()`
3. `\AmpProject\AmpWP\Editor\EditorSupport::maybe_show_notice()`

In the first, it's checking if the post global is set and so it isn't having this problem. However, the second two are not checking for the `$post` global and so they could be causing problems.

The three cases here should all be checking if `get_current_screen()->is_block_editor` (instead of checking whether the `base` is `'post'`) rather than exclusively checking if `get_post()` is set since it could be that a plugin creates that global on another screen. This is actually being done today in the last one so my suggested change isn't relevant there.

So maybe this is a better suggested alternative:

```diff
diff --git a/includes/admin/class-amp-post-meta-box.php b/includes/admin/class-amp-post-meta-box.php
index 86998c170..83e9955db 100644
--- a/includes/admin/class-amp-post-meta-box.php
+++ b/includes/admin/class-amp-post-meta-box.php
@@ -215,6 +215,11 @@ public function enqueue_admin_assets() {
 	 * @since 1.0
 	 */
 	public function enqueue_block_assets() {
+		$screen = get_current_screen();
+		if ( ! $screen || empty( $screen->is_block_editor ) ) {
+			return;
+		}
+
 		$post = get_post();
 		if ( ! $post instanceof WP_Post || ! in_array( $post->post_type, AMP_Post_Type_Support::get_eligible_post_types(), true ) ) {
 			return;
diff --git a/includes/validation/class-amp-validation-manager.php b/includes/validation/class-amp-validation-manager.php
index 402ec4070..779e958df 100644
--- a/includes/validation/class-amp-validation-manager.php
+++ b/includes/validation/class-amp-validation-manager.php
@@ -2175,6 +2175,11 @@ public static function enqueue_block_validation() {
 			return;
 		}
 
+		$screen = get_current_screen();
+		if ( ! $screen || empty( $screen->is_block_editor ) ) {
+			return;
+		}
+
 		$editor_support = Services::get( 'editor.editor_support' );
 
 		// Block validation script uses features only available beginning with WP 5.6.
```

In `\AmpProject\AmpWP\Editor\EditorSupport::maybe_show_notice()` it's checking `is_gutenberg_page()` which may not be relevant anymore.

Reply:
> The three cases here should all be checking if `get_current_screen()->is_block_editor` (instead of checking whether the base is 'post') rather than exclusively checking if `get_post()` is set since it could be that a plugin creates that global on another screen.

Since all those cases are using `\AmpProject\AmpWP\Editor\EditorSupport::editor_supports_amp_block_editor_features()` would you agree that that would be the most correct place to modify? That is:

```diff
diff --git a/src/Editor/EditorSupport.php b/src/Editor/EditorSupport.php
--- a/src/Editor/EditorSupport.php	(revision 38de6f6f86ca13e7c834e5688c4fa7144fc377b6)
+++ b/src/Editor/EditorSupport.php	(date 1628896711900)
@@ -91,6 +91,7 @@
 	 * @return bool
 	 */
 	public function editor_supports_amp_block_editor_features() {
-		return $this->dependency_support->has_support() && use_block_editor_for_post( get_post() );
+		$screen = get_current_screen();
+		return $this->dependency_support->has_support() && $screen && $screen->is_block_editor;
 	}
 }
```



> In `\AmpProject\AmpWP\Editor\EditorSupport::maybe_show_notice()` it's checking `is_gutenberg_page()` which may not be relevant anymore.

That's only relevant to users using Gutenberg v5.5 and older, where it's likely that their instance of WordPress (WP < v5) does not have `get_current_screen()->is_block_editor`.

Reply:
> Since all those cases are using `\AmpProject\AmpWP\Editor\EditorSupport::editor_supports_amp_block_editor_features()` would you agree that that would be the most correct place to modify?

I think so, but probably good to use `! empty( $screen->is_block_editor )` to prevent possible notices on older versions of WP.

Nevertheless, if this is done the phpdoc for the method should be updated to indicate that it is not only whether the editor in the “current environment” supports plugin features, but _also_ that the current screen is the block editor.

Maybe the method name should be changed to reflect this.

> That's only relevant to users using Gutenberg v5.5 and older, where it's likely that their instance of WordPress (WP < v5) does not have `get_current_screen()->is_block_editor`.

Correct, as we don't fully support versions older than WP 5.6.

Reply:
Any concerns on also checking if the post type is eligible for AMP in the same method (`EditorSupport::editor_supports_amp_block_editor_features()`)?

Reply:
This logic, yeah?

```php
if ( ! in_array( get_post_type(), AMP_Post_Type_Support::get_supported_post_types(), true ) ) {
	return;
}
```

Yeah, that seems like a good move because that same logic is duplicated currently in `\AMP_Post_Meta_Box::enqueue_block_assets()` and `\AmpProject\AmpWP\Editor\EditorSupport::maybe_show_notice()`:

https://github.com/ampproject/amp-wp/blob/041cc29a4c1baa834cb6bab62837e839f1f04858/includes/admin/class-amp-post-meta-box.php#L217-L221

https://github.com/ampproject/amp-wp/blob/041cc29a4c1baa834cb6bab62837e839f1f04858/src/Editor/EditorSupport.php#L63-L65

The logic is absent however in `\AMP_Validation_Manager::enqueue_block_validation()`. Indeed, I just disabled AMP for the Page post type, and erroneously the AMP toggle and AMP sidebar show up on that screen:

https://user-images.githubusercontent.com/134745/129429759-ec858f95-ba47-4994-8002-6453c5618e34.mov

This is fixed by adding the above condition. 

This also extends to `AMP_Post_Meta_Box::enqueue_block_assets()` as I can see it is currently using this condition:

https://github.com/ampproject/amp-wp/blob/041cc29a4c1baa834cb6bab62837e839f1f04858/includes/admin/class-amp-post-meta-box.php#L218-L221

Note how it's using `AMP_Post_Type_Support::get_eligible_post_types()` when it _should_ be using `AMP_Post_Type_Support::get_supported_post_types()`, as a post can be eligible without being actually supported (e.g. by being disabled). So in the above example with the `page` post type disabled, I'm erroneously getting `amp-block-editor` enqueued onto the page even though it is entirely disabled.

This is also seems to be a bug in `\AmpProject\AmpWP\Editor\EditorSupport::maybe_show_notice()`:

https://github.com/ampproject/amp-wp/blob/041cc29a4c1baa834cb6bab62837e839f1f04858/src/Editor/EditorSupport.php#L63-L65

It seems this should also rather be using `AMP_Post_Type_Support::get_supported_post_types()`.